docs/testplan/transceiver_onboarding_test_plan.md

```diff
@@ -0,0 +1,436 @@
+# Transceiver Onboarding Test Plan
+
+## Scope
+
+This test plan outlines a comprehensive framework for ensuring feature parity for new transceivers being onboarded to SONiC. The goal is to automate all tests listed in this document, covering the following areas:
+
+- **Link Behavior**: Test link behavior using shut/no shut commands and under process crash scenarios.
+- **Transceiver Information Fields**: Verify transceiver specific fields (Vendor name, part number, serial number) via CLI commands, ensuring values match expectations.
+- **Firmware**: Check firmware version readability and compliance with vendor-suggested values, using regex for version pattern matching.
+- **DOM Data**: Ensure Digital Optical Monitoring (DOM) data is correctly read and within acceptable ranges.
+- **Flags and Alerts**: Confirm no unexpected flags (e.g., Loss of Signal (LOS), Loss of Lock (LOL), DOM warnings) are set.
+- **Firmware Management**: Test firmware upgrade under various scenarios.
+- **Remote Reseat**: Verify support for remote reseat functionality.
```

Comment:
we should also consider reboot tests, make sure the cable up after reboot.

Reply:
do we test OIR behavior here?

Reply:
do we also check temperature of the transceivers?

Reply:
@lguohan
`we should also consider reboot tests, make sure the cable up after reboot.`
[MP] I have update the scope section to include device reboot scenarios.

`do we test OIR behavior here?`
[MP] Not currently. We have initiated conversation with vendor to help with pseudo OIR support.

`do we also check temperature of the transceivers?`
[MP] Yes, we are checking for temperature of transceivers to ensure it is within the temperature warning threshold.